tr improvement + dojo-like output

maxaalexeeva opened the pull request:
- tr improments from @maxaalexeeva, @alicekwak, and @sarahstueve 

Major changes: 
- a script to jointly extract mentions from papers and markdown files related to one model and assemble them into a json object similar to that produced by dojo api
- location and date rules
- rules and methods for command extraction and assembly of command sequences from command components extracted with rules
- adjusted python script for querying HomeController endpoints

Minor changes:
- new and adjusted rules 
- rearranging taxonomy and master ymls a bit
- storing file names in location attachment
- disallowing cosmos block merge if content starts with a digit---those are frequently just section ID---that is the start of a new section
- fixing a special char remapping bug
- added some methods for cross sentence event mentions
- adding color to commands in the visualizer

alicekwak commented on automates/text_reading/src/main/scala/org/clulab/aske/automates/data/DataLoader.scala:
```diff
@@ -123,8 +131,11 @@ class MarkdownTextDataLoader extends DataLoader {
-  def loadFile(f: File): Seq[String] = getTextFromFile(f).split("\n").filter(_.nonEmpty)
+  def loadFile(f: File): Seq[String] = getTextFromFile(f).split("\n").filter(_.nonEmpty).map(t=>t.replace("`", ""))
   override val extension: String = "md"
+  // split on #
```
Are these notes needed? They don't seem to explain anything. If they're for the future reference, make the notes into todo or fixme??

maxaalexeeva replied:
deleted

alicekwak commented on automates/text_reading/src/test/scala/org/clulab/aske/automates/text/TestParameterSetting.scala:
```diff
@@ -257,7 +257,7 @@ class TestParameterSetting  extends ExtractionTest {
   passingTest should s"extract the parameter setting(s) from u5a: ${u5a}" taggedAs(Somebody) in {
     val desired = Seq(
-      "value of shoot_lag" -> Seq("40") // fixme: ideally, should just be shoot_lag, but this tests the var-assigned rule
+      "shoot_lag" -> Seq("40") // fixme: ideally, should just be shoot_lag, but this tests the var-assigned rule
```
Seems like you addressed this fixme! Please delete the note before merging.

maxaalexeeva replied:
good eye! deleted

alicekwak commented on automates/text_reading/webapp/app/controllers/HomeController.scala:
```diff
@@ -248,7 +248,7 @@ class HomeController @Inject()(cc: ControllerComponents) extends AbstractControl
 
       for (m <- menInTextBlocks) {
-        val newMen = m.withAttachment(new MentionLocationAttachment(pageNum, blockIdx, "MentionLocation"))
+        val newMen = m.withAttachment(new MentionLocationAttachment("Fixme: pass filename", pageNum, blockIdx, "MentionLocation"))
```
What is that "Fixme: pass filename" for? Is that a message shown when there's some kind of problem with the file name??

maxaalexeeva replied:
argh I forgot about that one. I am having trouble fixing it. I will try before I request re-review. 

Basically, I am now storing filename in attachments and I am having trouble and I am having trouble accessing filename in this method.

alicekwak replied:
Okay, please let me know if you need any help with this!

alicekwak commented on automates/text_reading/webapp/public/javascripts/main.js:
```diff
@@ -114,7 +116,27 @@ var collData = {
+                             ]
+                           },
+       {
+                      "type": "CommandLineParamValuePair",
```
Is this indentation correct?

maxaalexeeva replied:
adjusted

alicekwak reviewed (changes requested):
Hi @maxaalexeeva! I finished reviewing your draft for the first round. Most of them looked awesome, but when I think something needs to be modified and/or reviewed I left comments. Please review them when you have some time. Also, as this work includes some of my work too, it would be great if you could review this PR too. Thanks!

maxaalexeeva commented on automates/text_reading/src/main/resources/application.conf:
```diff
@@ -141,7 +142,7 @@ alignment {
 modelComparisonAlignment {
   alignerType = "pairwisew2v"
-  w2vPath = "vectors.txt"
+  w2vPath = ${apps.projectDir}/automates/text_reading/src/main/resources/vectors.txt
```
@alicekwak I think this is the line that I changed to make sure that our vectors don't need to be copied to multiple places (for main vs text), so you should now store them under main/resources.

alicekwak replied:
Got it, thank you for letting me know!

maxaalexeeva commented on automates/text_reading/src/main/resources/org/clulab/aske_automates/grammars/dates.yml:
```diff
@@ -0,0 +1,29 @@
+    pattern: |
+      [word = /19\d\d/ & entity = "DATE"] | [word = /20\d\d/ & entity = "DATE"]
+
+  - name: date-event
```
I don't think I have added year anywhere, but it probably should be at some point? Either way, adding year under date---good idea!

maxaalexeeva commented on automates/text_reading/src/main/resources/org/clulab/aske_automates/grammars/markdown/commands.yml:
```diff
@@ -4,12 +4,125 @@ rules:
 ####COMMANDS####
 
-  - name: sample-command
+#   - name: sample-command
```
deleting

maxaalexeeva commented on automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala:
```diff
@@ -490,23 +524,149 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+      m match {
+        case m: TextBoundMention => {
+
+          // todo: should I filter out tbm that look like identifiers but are locations?
```
I tried it a couple of weeks ago and I have doubts---some identifiers are labeled as locations by the NER and we lose identifiers... I am inclined to not filter that way for now, but keep in mind that it might be reasonable thing to filter in future.

maxaalexeeva commented on automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala:
```diff
@@ -490,23 +524,149 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+  }
+
+
+  // sentence for debugging Bearing in mind , we set our default setting to the refugee move speed is equal to 200 km per day and the awareness of surrounding is 1 link .
```
@alicekwak .distinct won't work here because they are not identical mentions :( I have a method for doing distinct by text. I will see if I can use it for param settings.

alicekwak commented on automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala:
```diff
@@ -612,26 +610,26 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
   def locationsAreNotVariablesOrModels(mentions: Seq[Mention], state: State = new State()): Seq[Mention] = {
     // makes sure there are no mentions where variables are actually most likely locations
-    // for now only applied to descriptions, but may want to extend to other mentions
+    // currently not used because NER labels identifiers as LOCATION
```
Were there many cases where identifiers were labeled as Location?

maxaalexeeva replied:
at least two within our tests if I remember correctly.

alicekwak commented on automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinEngine.scala:
```diff
@@ -58,6 +58,7 @@ class OdinEngine(
   }
 
   var loadableAttributes = LoadableAttributes()
+  val actions = loadableAttributes.actions
```
Good thinking!

alicekwak commented on automates/text_reading/src/main/scala/org/clulab/aske/automates/apps/ExtractAndAssembleMentionEvents.scala:
```diff
@@ -382,7 +247,6 @@ object ExtractAndAssembleMentionEvents extends App {
           val paramSetObjs = new ArrayBuffer[ujson.Value]()
```
This "ParameterSetting" doesn't include "IntervalParameterSetting", right?

maxaalexeeva replied:
Correct. I need to add that...

alicekwak commented on automates/text_reading/src/main/scala/org/clulab/utils/MentionUtils.scala:
```diff
@@ -0,0 +1,116 @@
+  def getMentionText(mention: Mention): String = {
+
+    // get text bound mention text taking into account the presence of discontinuous character offset attachment
+    // fixme: add functionality to get text of an event mention whose arguments have discont attachments
```
Did you already address this fixme, or is it for future reference? If you did, please delete the note.

alicekwak reviewed (commented):
All the changes look great! There were only a few notes (mostly questions) this time. Please review them when you have some time. After they're addressed, I think it'll be good to be merged!

maxaalexeeva commented:
hey @alicekwak, I added int param setting output to the mention assembly json in response to your question and also modified a couple of python scripts that are supposed to run some of the endpoints in HomeController. I think I will stop with this branch modifications now :)

cl4yton reviewed (approved):
Heh, my ability to provide any meaningful review here is minimal :) 
What I do see is a *lot* of hard work!
Everything looks good to me!
